juriscraper/opinions/united_states/state/colo.py

```diff
@@ -23,25 +26,61 @@ def __init__(self, *args, **kwargs):
         self.url = "https://www.courts.state.co.us/Courts/Supreme_Court/Case_Announcements/"
 
     def _process_html(self) -> None:
-
         date = self.html.xpath(
             ".//p[contains(text(), 'case announcements for')]/text()"
-        )[0]
-        date_str = (
-            re.findall(r"case announcements for (.*) are now", date)[0]
-            .split(",", 1)[1]
-            .strip()
         )
-        for item in self.html.xpath(
-            ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
-        ):
-            docket, name = item.text_content().split(" ", 1)
-            self.cases.append(
-                {
-                    "date": date_str,
-                    "docket": docket,
-                    "name": name,
-                    "url": item.attrib["href"],
-                    "status": "Published",
-                }
+        if date:
+            date_str = (
+                re.findall(r"case announcements for (.*) are now", date[0])[0]
+                .split(",", 1)[1]
+                .strip()
             )
+            for item in self.html.xpath(
+                ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
+            ):
+                link_text = normalize_dashes(item.text_content().strip())
+                cite_match = re.findall(r"(\d{2,4}\s*CO\s*\d+\w?)", link_text)
+                docket_match = re.findall(r"\d+[A-Z]+\d{2,}", link_text)
+                docket = docket_match[0] if docket_match else ""
+                citation = cite_match[0] if cite_match else ""
+                name = (
+                    link_text.replace(docket, "")
+                    .replace(citation, "")
+                    .lstrip(",- ")
+                )
+                self.cases.append(
+                    {
+                        "date": date_str,
+                        "docket": docket,
+                        "name": name,
+                        "url": item.attrib["href"],
+                        "status": "Published",
+                        "citation": citation,
+                    }
+                )
+
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Pass scraped text into function and return data as a dictionary
+        Notes for 'Citation':
+            - Reporter key for this court: 'CO'
+            - Type for a state: 2
+        :param scraped_text: Text of scraped content
+        :return: metadata
+        """
+        cite_match = re.findall(r"\d{4}\s*CO\s*\d+", scraped_text)
+        docket_match = re.findall(r"\d+S[A-Z]\d+", scraped_text)
```

Comment:
Also- the regex pattern looks too broad here.

Reply:
Should I combine both RE's again @flooie ?

Reply:
Thats not what I was getting at.

first, I was suggesting we could tighten up the regex pattern  `r"\d+S[A-Z]\d+"` Surely we dont need [A-Z] here.  

Second I was suggesting we decouple the if statement so that if docket number or citation has at typo we dont just crash or return nothing. 


cite_match = re.findall(r"\d{4}\s*CO\s*\d+", scraped_text)
docket_match = re.findall(r"\d+S[A-Z]\d+", scraped_text)
metadata = {}
if cite_match:
    volume, page = cite_match[0].split("CO")
    metadata['Citation']['volume'] = volume
    metadata['Citation']['reporter'] = "CO"
    metadata['Citation']['page'] = page
if docket_match:
    metadata['OpinionCluster']['docket_number'] = docket_match[0]
```

Reply:
Ad back the search for the docket ina separate assignment.